### PR TITLE
Added DoctrineCache

### PR DESCRIPTION
**Pull request by slootjes**
Tests are included and I also included tests for the regular ArrayCache since it was missing.

### CODE REVIEW

**Comment by mpociot** [comment]
That's great! Thank you!

Could you also add `doctrine/cache` to the `suggest` part of the composer.json?
And we should include this somewhere in the README I guess :)

**Comment by slootjes** [comment]
Done. I also made a modification to the BotManFactory so cache and request are created automatically when not provided which makes it easier to start using the bot. I hope you agree :)

File: src/Mpociot/BotMan/BotManFactory.php [hunk 1]
```diff
@@ -18,8 +18,15 @@ class BotManFactory
      * @param CacheInterface $cache
      * @return \Mpociot\BotMan\BotMan
      */
-    public static function create(array $config, Request $request, CacheInterface $cache)
+    public static function create(array $config, Request $request, CacheInterface $cache = null)
```
Comment (mpociot) [inline src/Mpociot/BotMan/BotManFactory.php]: `$request` needs to be null as well, otherwise it can't be empty

**Comment by mpociot** [comment]
👍  Thank you very much.

**Comment by slootjes** [comment]
You're welcome :)

**Pull request merged**